driver/src/contracts/stablex_contract.rs

```diff
@@ -59,12 +59,74 @@ impl StableXContract for StableXContractImpl {
 
     fn submit_solution(
         &self,
-        _batch_index: U256,
-        _orders: Vec<Order>,
-        _solution: Solution,
+        batch_index: U256,
+        orders: Vec<Order>,
+        solution: Solution,
     ) -> Result<()> {
-        unimplemented!();
+        let account = self
+            .base
+            .account_with_sufficient_balance()
+            .ok_or("Not enough balance to send Txs")?;
+
+        let (owners, order_ids, volumes, prices, token_ids_for_price) =
+            parse_auction_results(orders, solution);
+
+        self.base
+            .contract
+            .call(
+                "submitSolution",
+                (
+                    batch_index,
+                    owners,
+                    order_ids,
+                    volumes,
+                    prices,
+                    token_ids_for_price,
+                ),
+                account,
+                Options::default(),
+            )
+            .wait()
+            .map_err(DriverError::from)
+            .map(|_| ())
+    }
+}
+
+type ContractRecognizedAuctionResults = (Vec<H160>, Vec<U128>, Vec<U128>, Vec<U128>, Vec<U128>);
+
+fn parse_auction_results(
+    orders: Vec<Order>,
+    solution: Solution,
+) -> ContractRecognizedAuctionResults {
+    // Representing the solution's price vector more compactly as:
+    // sorted_touched_token_ids, non_zero_prices which are logically bound by index.
+    // Example solution.prices = [3, 0, 1] will be transformed into [0, 2], [3, 1]
+    let mut ordered_token_ids: Vec<U128> = vec![];
+    let mut prices: Vec<U128> = vec![];
+    for (token_id, price) in solution.prices.into_iter().enumerate() {
+        if price > 0 {
+            ordered_token_ids.push(U128::from(token_id as usize));
+            prices.push(U128::from(price as usize));
+        }
+    }
+
+    let mut owners: Vec<H160> = vec![];
+    let mut order_ids: Vec<U128> = vec![];
+    let mut volumes: Vec<U128> = vec![];
+    let zipped_amounts = solution
+        .executed_buy_amounts
+        .into_iter()
+        .zip(solution.executed_sell_amounts.into_iter());
+    for (order_id, (buy_amount, sell_amount)) in zipped_amounts.enumerate() {
```

Comment:
This information comes as part of the auction data. We might be able to use the BatchInformation struct on the Order model for storing the index?

Reply:
Good points here indeed. I have also been thinking that our existing `Order` struct is slightly insufficient for the requirements of the StableX Order type. Observe the comparison, 

StableX
```js
struct Order {
    uint16 buyToken;
    uint16 sellToken;
    uint32 validFrom;  // order is valid from auction collection period: validFrom inclusively
    uint32 validUntil;  // order is valid till auction collection period: validUntil inclusively
    bool isSellOrder;
    uint128 priceNumerator;
    uint128 priceDenominator;
    uint128 remainingAmount; // remainingAmount can either be a sellAmount or buyAmount, depending on the flag isSellOrder
}
```

SnappAuction

```js
SellOrder(
    uint auctionId,
    uint16 slotIndex,
    uint16 accountId,
    uint8 buyToken,
    uint8 sellToken,
    uint96 buyAmount,
    uint96 sellAmount
)
```

There are actually very few values which coincide here. Even those that do have differing types (`u8` vs `u16`). Perhaps it would be in our best interest to introduce a new Order Type for the StableX to interact with? @josojo, @fleupold and/or  @anxolin any suggestions? At this point I see very little similarities. Should we try and make this a single (robust) Order type or is it really worth it?

Reply:
>Perhaps it would be in our best interest to introduce a new Order Type for the StableX to interact with?

I guess so.

Reply:
I am already working on another branch implementing `get_auction_info` which will explicitly deal with `Orders::from_bytes`. This will be a more relevant place to make the changes.

Reply:
I'm not sure if we need different models. isSellOrder will be relevant for the dfusion order as well (and since right now it's always true, I'd ignore it). `priceNumerator` `priceDenominator` and `remainingAmount` can be used to compute buyAmount and sellAmount.

The most shoehorning I see with using `slotIndex` vs `orderID` interchangeably on the same model (since in fact they are different things).

If we create a new model, we will also have to implement the price finder in a way that can handle both subtypes, which might open up a few other can of worms down the road. This is why I'd hope we can share the data model (as at the end of the day our orders didn't change that much)

Reply:
Thats cool for now, then I will use slotIndex for orderId in this PR and deal with whatever else is necessary when I get to the `get_auction_data`. Cool with everyone else here?

Reply:
We should be careful to note that `slot_index` (from and Order's BatchInformation) is a `u16` which could result in future issues. I will include a comment about this in the code.